src/common/fs_types.cc

```diff
@@ -86,10 +87,12 @@ void file_layout_t::encode(ceph::buffer::list& bl, uint64_t features) const
   }
 
   ENCODE_START(2, 2, bl);
-  encode(stripe_unit, bl);
-  encode(stripe_count, bl);
-  encode(object_size, bl);
-  encode(pool_id, bl);
+  encode(std::tuple{
+    stripe_unit,
+    stripe_count,
+    object_size,
+    pool_id,
+  }, bl, 0);
   encode(pool_ns, bl);
```

Comment:
why was this left behind? does this specifically need separate encoding?

Reply:
It's a `std::string` and (a) doesn't benefit from this optimization and (b) causes this to become variable-length; that may undo the optimization.
In this PR, I merged calls only for adjacent trivial fixed-size types.

Reply:
OK, added to commit messages